app/react/src/client/testing/index.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  composeStory as originalComposeStory,
+  composeStories as originalComposeStories,
+} from '@storybook/store';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { composeConfigs } from '@storybook/preview-web';
+import { render } from '../preview/render';
+
+import type { Meta, ReactFramework } from '../preview/types-6-0';
+import type { StoriesWithPartialProps, GlobalConfig, StoryFile, TestingStory } from './types';
+
+const defaultGlobalConfig: GlobalConfig = {
+  render,
+};
+
+let globalStorybookConfig = {
+  ...defaultGlobalConfig,
+};
+
+/** Function that sets the globalConfig of your storybook. The global config is the preview module of your .storybook folder.
+ *
+ * It should be run a single time, so that your global config (e.g. decorators) is applied to your stories when using `composeStories` or `composeStory`.
+ *
+ * Example:
+ *```jsx
+ * // setup.js (for jest)
+ * import { setGlobalConfig } from '@storybook/testing-react';
+ * import * as globalStorybookConfig from './.storybook/preview';
+ *
+ * setGlobalConfig(globalStorybookConfig);
+ *```
+ *
+ * @param config - e.g. (import * as globalConfig from '../.storybook/preview')
+ */
+export function setGlobalConfig(config: GlobalConfig) {
+  globalStorybookConfig = composeConfigs([defaultGlobalConfig, config]) as GlobalConfig;
+}
+
+export function composeStory<GenericArgs>(
+  story: TestingStory<GenericArgs>,
+  meta: Meta<GenericArgs | any>,
+  globalConfig: GlobalConfig = globalStorybookConfig
+) {
+  const projectAnnotations = { ...defaultGlobalConfig, ...globalConfig };
+
+  return originalComposeStory<ReactFramework, GenericArgs>(story, meta, projectAnnotations);
+}
+
+export function composeStories<TModule extends StoryFile>(
+  storiesImport: TModule,
+  globalConfig?: GlobalConfig
+) {
+  const composedStories = originalComposeStories(storiesImport, globalConfig, composeStory);
```

Comment:
It's handled inside of `composeStory`, so it will always be used regardless if the users use `composeStories` or `composeStory` directly

Reply:
Are you sure @yannbf? Because in my reading, `composeStory` calls `originalComposeStory` with the `defaultGlobalConfig`, whereas `composeStories` calls `originalComposeStories`, which calls `originalComposeStory` *but not* `composeStory`.

Reply:
Maybe the code you saw was outdated?

```
const composedStories = originalComposeStories(storiesImport, globalConfig, composeStory);
```